### DISCUSSION TITLE
WLANPi Pro as a Wi-Fi client

### DISCUSSION THREAD

**Discussion by MaciejStawiar** [WLAN-Pi/feedback#32] [General Inquiries] [answered]
Hi All,
Do You have a method to use one of the AX210 as a Wi-Fi WPA3 client?
I tried so far editing /etc/wpa_supplicant.conf file and installing network manager suite to run nmtui or nmcli but no success.
Goal is to run iPerf tests against 6E AP from Pro. 
Or maybe Ubuntu Core image on a separate card will work?

**Comment by jiribrejcha** [comment]
Hi Maciej,

This should work just fine to connect the AX210 client to an AP.

Modify the file:
`sudo nano /etc/wpa_supplicant/wpa_supplicant.conf`
Add this to the very bottom of the file:
```
network={
  ssid="6ghz"
  psk="wlanpi123"
  priority=10
  key_mgmt=SAE
  ieee80211w=2
}
```

Connect the client to AP using this command. On the Pro we would use wlan1 client interface, and wlan0 on other WLAN Pis:
`sudo wpa_supplicant -c /etc/wpa_supplicant/wpa_supplicant.conf -i wlan1`

**Reply by MaciejStawiar** [reply]
So, still no success, on the same SSID my Win11 powered AX210 works fine. 

What I did: I flashed the fresh image and proceed - wpa supplicant looks like it is working

> 
wlanpi@wlanpi-095:~ $ sudo wpa_supplicant -c /etc/wpa_supplicant/wpa_supplicant.conf -i wlan1
[sudo] password for wlanpi:
Successfully initialized wpa_supplicant
wlan1: SME: Trying to authenticate with b0:27:cf:7f:e1:74 (SSID='6GHz' freq=6135 MHz)
wlan1: CTRL-EVENT-AUTH-REJECT b0:27:cf:7f:e1:74 auth_type=3 auth_transaction=1 status_code=1
wlan1: SME: Trying to authenticate with b0:27:cf:7f:e1:74 (SSID='6GHz' freq=6135 MHz)
wlan1: CTRL-EVENT-AUTH-REJECT b0:27:cf:7f:e1:74 auth_type=3 auth_transaction=1 status_code=1
wlan1: SME: Trying to authenticate with b0:27:cf:7f:e1:74 (SSID='6GHz' freq=6135 MHz)
wlan1: CTRL-EVENT-AUTH-REJECT b0:27:cf:7f:e1:74 auth_type=3 auth_transaction=1 status_code=1
wlan1: SME: Trying to authenticate with b0:27:cf:7f:e1:74 (SSID='6GHz' freq=6135 MHz)
wlan1: CTRL-EVENT-AUTH-REJECT b0:27:cf:7f:e1:74 auth_type=3 auth_transaction=1 status_code=1
wlan1: CTRL-EVENT-SSID-TEMP-DISABLED id=0 ssid="6GHz" auth_failures=1 duration=10 reason=CONN_FAILED
wlan1: CTRL-EVENT-SSID-REENABLED id=0 ssid="6GHz"
wlan1: SME: Trying to authenticate with b0:27:cf:7f:e1:74 (SSID='6GHz' freq=6135 MHz)
wlan1: CTRL-EVENT-AUTH-REJECT b0:27:cf:7f:e1:74 auth_type=3 auth_transaction=1 status_code=1
BSSID b0:27:cf:7f:e1:74 ignore list count incremented to 2, ignoring for 10 seconds
wlan1: CTRL-EVENT-SSID-TEMP-DISABLED id=0 ssid="6GHz" auth_failures=2 duration=20 reason=CONN_FAILED
wlan1: CTRL-EVENT-SSID-REENABLED id=0 ssid="6GHz"
wlan1: SME: Trying to authenticate with b0:27:cf:7f:e1:74 (SSID='6GHz' freq=6135 MHz)
wlan1: CTRL-EVENT-AUTH-REJECT b0:27:cf:7f:e1:74 auth_type=3 auth_transaction=1 status_code=1
wlan1: CTRL-EVENT-SSID-TEMP-DISABLED id=0 ssid="6GHz" auth_failures=3 duration=30 reason=CONN_FAILED
wlan1: CTRL-EVENT-SSID-REENABLED id=0 ssid="6GHz"
wlan1: SME: Trying to authenticate with b0:27:cf:7f:e1:74 (SSID='6GHz' freq=6135 MHz)
wlan1: CTRL-EVENT-AUTH-REJECT b0:27:cf:7f:e1:74 auth_type=3 auth_transaction=1 status_code=1
wlan1: CTRL-EVENT-SSID-TEMP-DISABLED id=0 ssid="6GHz" auth_failures=4 duration=60 reason=CONN_FAILED
wlan1: CTRL-EVENT-SSID-REENABLED id=0 ssid="6GHz"
wlan1: SME: Trying to authenticate with b0:27:cf:7f:e1:74 (SSID='6GHz' freq=6135 MHz)
wlan1: CTRL-EVENT-AUTH-REJECT b0:27:cf:7f:e1:74 auth_type=3 auth_transaction=1 status_code=1
wlan1: CTRL-EVENT-SSID-TEMP-DISABLED id=0 ssid="6GHz" auth_failures=5 duration=60 reason=CONN_FAILED
^Cwlan1: CTRL-EVENT-DSCP-POLICY clear_all
wlan1: CTRL-EVENT-DSCP-POLICY clear_all
nl80211: deinit ifname=wlan1 disabled_11b_rates=0
wlan1: CTRL-EVENT-TERMINATING

> 

I see that problem is with authentication, so decided to capture some frames using remote capture from Wireshark on wlan0.

my wlan1 mac:

> 

wlan1: flags=4099<UP,BROADCAST,MULTICAST>  mtu 1500
        ether 84:14:4d:a7:85:5f  txqueuelen 1000  (Ethernet)

> 

I can't attach pcap so will paste some print screens:
1. overall view with the filter applied, 1st Auth frame Successfull:
![pcap1](https://user-images.githubusercontent.com/44684683/197392924-5cd961b8-6ccf-4223-86ec-1133a9a36e37.png)

2. 2nd shows unspecified failure, so not very useful:
![pcap2](https://user-images.githubusercontent.com/44684683/197393064-84c3a327-3d5e-4438-aa7e-603cdc7cdaa1.png)

And this sequence repeats as many times as can be seen in wpa_supplicant output.
There are no any other types of frames (on this filter).

Any ideas what to try next?
Thanks for help in advance :)

**Comment by jiribrejcha** [comment]
If you are using WPA3 SAE, you might need to add this line to /etc/wpa_supplicant/wpa_supplicant.conf:
`sae_pwe=1`

**Reply by faisalzkhan** [reply]
I have a wlan pi pro, I wanted it to connect t 6GHZ ssid, WPA3 SAE. I have added this sae_pwe=1 in file /etc/wpa_supplicant/wpa_supplicant.conf
But its not working, I am getting error:

root@wlanpi-09b:/etc/wpa_supplicant# wpa_supplicant -c /etc/wpa_supplicant/wpa_supplicant.conf -i wlan1 &
[1] 1077437
root@wlanpi-09b:/etc/wpa_supplicant# Successfully initialized wpa_supplicant
Line 3: unknown global field 'sae_pwe=1'.
Line 3: Invalid configuration line 'sae_pwe=1'.
Failed to read or parse configuration '/etc/wpa_supplicant/wpa_supplicant.conf'.

[1]+  Exit 255                wpa_supplicant -c /etc/wpa_supplicant/wpa_supplicant.conf -i wlan1
root@wlanpi-09b:/etc/wpa_supplicant#

**Comment by MaciejStawiar** [comment, marked as answer]
Successs! I have added this line to wpa_supplicant.conf and then run dhclient to get ipaddress.

So for future reference here is my wpa_supplicant.conf (last most important part):

```
sae_pwe=1

network={
  ssid="YOUR_SSID"
  psk="YOUR_PASSPHRASE"
  priority=10
  key_mgmt=SAE
  ieee80211w=2
}
```

wpa_supplicant output:

```
wlanpi@wlanpiPro:~ $ sudo wpa_supplicant -c /etc/wpa_supplicant/wpa_supplicant.conf -i wlan1
Successfully initialized wpa_supplicant
wlan1: CTRL-EVENT-REGDOM-CHANGE init=DRIVER type=COUNTRY alpha2=KR
wlan1: SME: Trying to authenticate with b0:27:cf:7f:e1:74 (SSID='6GHz' freq=6135 MHz)
wlan1: SME: Trying to authenticate with b0:27:cf:7f:e1:74 (SSID='6GHz' freq=6135 MHz)
wlan1: PMKSA-CACHE-ADDED b0:27:cf:7f:e1:74 0
wlan1: Trying to associate with b0:27:cf:7f:e1:74 (SSID='6GHz' freq=6135 MHz)
wlan1: Associated with b0:27:cf:7f:e1:74
wlan1: CTRL-EVENT-SUBNET-STATUS-UPDATE status=0
wlan1: WPA: Key negotiation completed with b0:27:cf:7f:e1:74 [PTK=CCMP GTK=CCMP]
wlan1: CTRL-EVENT-CONNECTED - Connection to b0:27:cf:7f:e1:74 completed [id=0 id_str=]
```

then run this command to get ip address from dhcp:
```
wlanpi@wlanpiPro:~ $ sudo dhclient wlan1
```

FPMS shows that there is no ip address, but ifconfig and ping says different:

```
wlan1: flags=4163<UP,BROADCAST,RUNNING,MULTICAST>  mtu 1500
        inet 10.0.0.32  netmask 255.255.255.0  broadcast 10.0.0.255
        inet6 fe80::489e:96a4:6abb:1770  prefixlen 64  scopeid 0x20<link>
        ether 84:14:4d:a7:85:5f  txqueuelen 1000  (Ethernet)
        RX packets 294  bytes 119399 (116.6 KiB)
        RX errors 0  dropped 0  overruns 0  frame 0
        TX packets 192  bytes 23098 (22.5 KiB)
        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0

wlanpi@wlanpiPro:~ $ ping 10.0.0.1
PING 10.0.0.1 (10.0.0.1) 56(84) bytes of data.
64 bytes from 10.0.0.1: icmp_seq=1 ttl=64 time=1.82 ms
64 bytes from 10.0.0.1: icmp_seq=2 ttl=64 time=1.46 ms
64 bytes from 10.0.0.1: icmp_seq=3 ttl=64 time=2.18 ms
^C
--- 10.0.0.1 ping statistics ---
3 packets transmitted, 3 received, 0% packet loss, time 2003ms
rtt min/avg/max/mdev = 1.455/1.817/2.175/0.293 ms
```

Thanks Jiri, cheers!
